modules/core/src/persistence.cpp

```diff
@@ -7375,28 +7375,35 @@ void read(const FileNode& node, int& value, int default_value)
 {
     value = !node.node ? default_value :
     CV_NODE_IS_INT(node.node->tag) ? node.node->data.i :
-    CV_NODE_IS_REAL(node.node->tag) ? cvRound(node.node->data.f) : 0x7fffffff;
+    CV_NODE_IS_REAL(node.node->tag) ? std::numeric_limits<int>::max() : std::numeric_limits<int>::max();
```

Comment:
Looks like INT_MAX will be returned in both cases. If this is intended why we even bother to check whether node is REAL?